File: Niveau 3/Niveau 3.7/stock2.hs
--------------------------------------------------------------------------------
-- Name:        État du stock
-- Purpose:     France-IOI niveau 3.7
--
-- Author:      Sébastien
--
-- Created:     24/09/2022
--------------------------------------------------------------------------------

import Control.Monad
import Data.List

-- Applique une opération sur la ième valeur de la liste
modifier :: [Int] -> Int -> Int -> [Int]
modifier liste indice operation = 
    (take indice liste) ++ [liste!!indice + operation] ++ (drop (indice+1) liste)

traiter :: Int -> [Int] -> IO [Int]
traiter 0 stock = return stock
traiter q stock = do
    stock' <- traiter (q-1) stock
    (produit:operation:_) <- fmap (map read . words) getLine :: IO [Int]
    -- putStrLn $ "Opération #" ++ show q ++ ": produit " ++ show produit ++ " -> " ++ show operation
    return $ modifier stock' (produit-1) operation

main :: IO()
main = do
    -- Lit le stock initial
    n <- readLn :: IO Int
    stock <- fmap (map read . words) getLine :: IO [Int]
    
    -- Réalise les opérations sur le stock
    q <- readLn :: IO Int
    newStock <- traiter q stock
    
    -- Affiche le stock final
    putStrLn $ intercalate " " (map show newStock)
    

File: Niveau 3/Niveau 3.7/stock2.py
#-------------------------------------------------------------------------------
# Name:        État du stock
# Purpose:     France-IOI niveau 3.7
#
# Author:      Sébastien
#
# Created:     24/09/2022
#-------------------------------------------------------------------------------

import sys

def main():
    n = int(input())
    stock = list(map(int, sys.stdin.readline().split()))
    o = int(input())
    for _ in range(o):
        produit, operation = map(int, sys.stdin.readline().split())
        stock[produit-1] += operation
    print(" ".join(map(str, stock)))

if __name__ == '__main__':
    main()

"""
4
5 10 0 4
8
1 1
2 -2
4 2
3 5
3 -2
3 -1
1 4
1 -5
"""